eras/conway/impl/test/Test/Cardano/Ledger/Conway/GovSpec.hs

```diff
@@ -0,0 +1,128 @@
+                        constitutionHash
+                    )
+                    SNothing
+                )
+        submitFailingProposal constitutionActionNext
+      itM @Conway "Lookup snapshots" $ do
+        constitutionHash <- freshSafeHash
+        let constitutionAction =
+              NewConstitution
+                SNothing
+                ( Constitution
+                    ( Anchor
+                        (fromJust $ textToUrl "constitution.0")
+                        constitutionHash
+                    )
+                    SNothing
+                )
+        gaidConstitutionProp <- submitProposal constitutionAction
+        modifyNES $ \nes ->
+          nes
+            & nesEsL . esLStateL . lsUTxOStateL . utxosGovStateL . cgEnactStateL . ensPrevConstitutionL
+              .~ SJust (PrevGovActionId gaidConstitutionProp) -- Add the wrong one to PrevGovActionIds
+            & nesEsL . esLStateL . lsUTxOStateL . utxosGovStateL . cgGovSnapshotsL
+              .~ def -- Remove from snapshots
+        let constitutionActionNext =
+              NewConstitution
+                (SJust $ PrevGovActionId (gaidConstitutionProp {gaidGovActionIx = GovActionIx 1})) -- make it wrong
+                ( Constitution
+                    ( Anchor
+                        (fromJust $ textToUrl "constitution.1")
+                        constitutionHash
+                    )
+                    SNothing
+                )
+        submitFailingProposal constitutionActionNext
+    context "Valid proposals are accepted" $ do
+      itM @Conway "Lookup snapshots" $ do
+        constitutionHash <- freshSafeHash
+        let constitutionAction =
+              NewConstitution
+                SNothing
+                ( Constitution
+                    ( Anchor
+                        (fromJust $ textToUrl "constitution.0")
+                        constitutionHash
+                    )
+                    SNothing
+                )
+        gaidConstitutionProp <- submitProposal constitutionAction
+        let constitutionActionNext =
+              NewConstitution
+                (SJust $ PrevGovActionId gaidConstitutionProp)
+                ( Constitution
+                    ( Anchor
+                        (fromJust $ textToUrl "constitution.1")
+                        constitutionHash
+                    )
+                    SNothing
+                )
+        submitProposal constitutionActionNext
+      itM @Conway "Lookup PrevGovActionIds" $ do
+        constitutionHash <- freshSafeHash
+        let constitutionAction =
+              NewConstitution
+                SNothing
+                ( Constitution
+                    ( Anchor
+                        (fromJust $ textToUrl "constitution.0")
+                        constitutionHash
+                    )
+                    SNothing
+                )
+        gaidConstitutionProp <- submitProposal constitutionAction
```

Comment:
You should not have it in the proposals if you are testing that the id matches the previously enacted one. In other words in this test you have an impossible state: govActionid is both: the previously enacted one and one of the currently proposed gov actions

You should just generate `gaidConstitutionProp` randomly
```suggestion
```